packages/MainUI/components/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,274 @@
+          marginLeft: '0.2rem',
+          border: `1px solid ${theme.palette.baselineColor.transparentNeutral[30]}`,
+        };
+      }
+      if (btnId === BUTTON_IDS.REFRESH) {
+        return {
+          padding: '0.75rem',
+          maxHeight: '2rem',
+          background: theme.palette.baselineColor.neutral[100],
+          borderRadius: '0 6.25rem 6.25rem 0',
+          color: theme.palette.baselineColor.neutral[0],
+          '&:hover': {
+            background: theme.palette.dynamicColor.main,
+          },
+        };
+      }
+      return undefined;
+    };
+
+    return {
+      leftSection: {
+        buttons: buttons
+          .filter(btn => LEFT_SECTION_BUTTONS.includes(btn.id as StandardButtonId))
+          .map(btn => {
+            const config = createStandardButtonConfig(btn as StandardButton);
+            const style = standardButtonStyle(btn.id);
+            if (style) {
+              config.sx = style;
+            }
+            if (btn.id === BUTTON_IDS.NEW) {
+              config.iconText = btn.name;
+            }
+            return config;
+          }),
+        style: {
+          display: 'flex',
+          width: 'auto',
+          alignItems: 'center',
+          background: theme.palette.baselineColor.neutral[0],
+          borderRadius: '10rem',
+          padding: '0.25rem',
+          maxHeight: '2.5rem',
+          gap: '0.05rem',
+        },
+      },
+      centerSection: {
+        buttons: buttons
+          .filter(btn => CENTER_SECTION_BUTTONS.includes(btn.id as StandardButtonId))
+          .map(btn => createStandardButtonConfig(btn as StandardButton)),
+        style: {
+          display: 'flex',
+          width: '100%',
+          background: theme.palette.baselineColor.transparentNeutral[5],
+          borderRadius: '10rem',
+          padding: '.25rem',
+          gap: '0.25rem',
+        },
+      },
+      rightSection: {
+        buttons: buttons
+          .filter(btn => {
+            if (isProcessButton(btn)) return true;
+            return RIGHT_SECTION_BUTTONS.includes(btn.id as StandardButtonId);
+          })
+          .map(btn => {
+            if (isProcessButton(btn)) {
+              const config = createProcessButtonConfig(btn);
+              return {
+                ...config,
+                customComponent: () => (
+                  <GenericProcessButton
+                    button={btn}
+                    onClick={config.onClick}
+                    disabled={!recordContext?.selectedRecord?.id}
+                  />
+                ),
+              };
+            }
+            return createStandardButtonConfig(btn as StandardButton);
+          }),
+        style: {
+          display: 'flex',
+          gap: '0.25rem',
+          background: theme.palette.baselineColor.transparentNeutral[5],
+          borderRadius: '10rem',
+          padding: '0.25rem',
+        },
+      },
+      isItemSelected: !!recordContext?.selectedRecord,
+    };
+  };
```

Comment:
is there any way you could simplify, refactor into smaller pieces and re-usable ones?